src/table/components/virtualized-table/hooks/use-infinite-scroll-data.ts

```diff
@@ -43,51 +43,143 @@ const createNewRow = (
   };
 };
 
+const isRowMissingData = (rows: Row[], x: number, y: number, width: number) => {
```

Comment:
this and the other two functions are pure functions right? Could they be in a util file or something? this file is getting pretty big

Reply:
Agree, they can be moved